include/wlr/types/wlr_idle.h

```diff
@@ -0,0 +1,48 @@
+#ifndef WLR_TYPES_WLR_IDLE_H
+#define WLR_TYPES_WLR_IDLE_H
+
+#include <wayland-server.h>
+
+/**
+ * Idle protocol is used to create timers which will notify the client when the
+ * compositor does not receive any input for a given time(in milliseconds). Also
+ * the client will be notify when the timer receve an activity notify and already
+ * was in idle state. Besides this, the client is able to simulate user activity
+ * which will reset the timers and at any time can destroy the timer.
+ */
+
+
+struct wlr_idle {
+	struct wl_global *wl_global;
+	struct wl_list idle_timers; // wlr_idle_timeout::link
+	struct wl_event_loop *event_loop;
+
+	struct wl_listener display_destroy;
+	struct wl_signal activity_notify;
+
+	void *data;
+};
+
+struct wlr_idle_timeout {
+	struct wl_resource *resource;
+	struct wl_list link;
+
+	struct wl_event_source *idle_source;
+	bool idle_state;
+	uint32_t timeout; // milliseconds
+
+	struct wl_listener input_listener;
+
+	void *data;
+};
+
+struct wlr_idle *wlr_idle_create(struct wl_display *display);
+
+void wlr_idle_destroy(struct wlr_idle *idle);
+
+/**
+ * Send notification to restart all timers. Called by compositor when
+ * there is an user activity event.
+ */
+void wlr_idle_notify_activity(struct wlr_idle *idle);
```

Comment:
The seat on which the activity needs to be notified is missing.

Reply:
I no longer need the seat now, because wlr_idle holds the wl_signal to notify the timers on user activity

Reply:
You need both the `wlr_idle` and the `wlr_seat`. When a client wants to be notified when a seat is idle, it doesn't care if another seat is not idle. In other words: when I move my mouse, the mouse is tied to a seat, and that should reset the idle timer for this seat only, not for all seats.